lbrycrd/stuff/start.sh

```diff
@@ -6,35 +6,39 @@
 # ## not specified it will only create an index for transactions that are related to the wallet or have unspent outputs.
 # ## This is specific to chainquery.
 
+# The config file does not exist in the container image. It must be mounted, or
+# if not, a default config is generated using environment variables.
+CONFIG_PATH=/etc/lbry/lbrycrd.conf
+if [ -f "$CONFIG_PATH" ]
+then
+    echo "Using the config file mounted into the container"
+else
+    echo "Creating fresh config file from environment variables"
+    ## Set config params
+    mkdir -p `dirname $CONFIG_PATH`
+    echo "rpcuser=$RPC_USER" > $CONFIG_PATH
+    echo "rpcpassword=$RPC_PASSWORD" >> $CONFIG_PATH
+    echo "rpcallowip=$RPC_ALLOW_IP" >> $CONFIG_PATH
+    echo "rpcport=9245" >> $CONFIG_PATH
+    echo "rpcbind=0.0.0.0" >> $CONFIG_PATH
+    #echo "bind=0.0.0.0" >> $CONFIG_PATH
+fi
+
 ## Ensure perms are correct prior to running main binary
-mkdir -p /data/.lbrycrd
 chown -R lbrycrd:lbrycrd /data
 chmod -R 755 /data/
 
-## TODO: Consider a config directory for future magic.
-# chown -R 1000:1000 /etc/lbrycrd
-# chmod -R 755 /etc/lbrycrd
 rm -f /var/run/lbrycrd.pid
 
-
-## Set config params
-## TODO: Make this more automagic in the future.
-echo "rpcuser=$RPC_USER" > /data/.lbrycrd/lbrycrd.conf
-echo "rpcpassword=$RPC_PASSWORD" >> /data/.lbrycrd/lbrycrd.conf
-echo "rpcallowip=$RPC_ALLOW_IP" >> /data/.lbrycrd/lbrycrd.conf
-echo "rpcport=9245" >> /data/.lbrycrd/lbrycrd.conf
-echo "rpcbind=0.0.0.0" >> /data/.lbrycrd/lbrycrd.conf
-#echo "bind=0.0.0.0" >> /data/.lbrycrd/lbrycrd.conf
-
 ## Control this invocation through envvar.
 case $RUN_MODE in
   default )
-    su -c "lbrycrdd -server -conf=/data/.lbrycrd/lbrycrd.conf -printtoconsole" lbrycrd
```

Comment:
So the one reason I kept this as the default behavior is when you run this script as root to launch it you gain the luxury of being able to enforce the file ownership and permissions of the config.  

You then de-escalate privileges in order to run the actual application and gain the same benefits of a unpriv'd user.

Reply:
happy to leave it the way it is if I can solve the `su : must be run from a terminal` problem I have.